components/sensor/ultrasonic/ultrasonic.go

```diff
@@ -149,8 +149,6 @@ func (s *Sensor) Readings(ctx context.Context, extra map[string]interface{}) (ma
 		return nil, errors.Wrap(err, "ultrasonic: error getting digital interrupt ticks")
 	}
 
-	// Remove the callbacks added by the interrupt stream once we are done reading.
-	defer echoInterrupt.RemoveCallback(s.ticksChan)
```

Comment:
Note to myself: it's okay to remove this line because the callbacks will be removed when `ctx` in canceled, which should happen when this function returns.